gitbutler-ui/src/lib/components/AISettings.svelte

```diff
@@ -0,0 +1,239 @@
+<script lang="ts">
+	import Select from './Select.svelte';
+	import SelectItem from './SelectItem.svelte';
+	import TextBox from './TextBox.svelte';
+	import { GitConfig } from '$lib/backend/gitConfig';
+	import {
+		AnthropicModel,
+		KeyOption,
+		ModelKind,
+		OpenAIModel,
+		SummarizerSettings
+	} from '$lib/backend/summarizerSettings';
+	import RadioButton from '$lib/components/RadioButton.svelte';
+	import SectionCard from '$lib/components/SectionCard.svelte';
+
+	export let summarizerSettings = new SummarizerSettings(new GitConfig());
+
+	let modelKind: ModelKind | undefined;
+	summarizerSettings.getModelKind().then((persistedModelKind) => (modelKind = persistedModelKind));
+	$: if (modelKind) summarizerSettings.setModelKind(modelKind);
+	$: if (form && modelKind) form.modelKind.value = modelKind;
+
+	let keyOption: { name: string; value: KeyOption } | undefined;
+	summarizerSettings
+		.getKeyOption()
+		.then(
+			(persistedKeyOption) =>
+				(keyOption = keyOptions.find((option) => option.value == persistedKeyOption))
+		);
+	$: if (keyOption) summarizerSettings.setKeyOption(keyOption.value);
```

Comment:
Do these run on first initialization atm?

Reply:
Yes. Is there a clever way to prevent this, perhaps with the onMount? I looked to see if there was a svelte-y way to track changes and the solutions I found suggested having a second variable you assigned the previous value to which seemed very cumbersome.

Reply:
Thinking about the AI service and how I want to interact with the RXJS, I might end up structuring them in such a way where this isn't a problem

Reply:
So, even after using a writable, I needed to add in an "initialized" check which I'm not super thrilled abouthttps://github.com/gitbutlerapp/gitbutler/pull/3084/commits/206d6cce07edfc4bf4d082b440423b8ad0da7844.

I thought registering the subscriber after the set call would have done the trick.

Reply:
I'll pull down your branch locally so I can have a better look.